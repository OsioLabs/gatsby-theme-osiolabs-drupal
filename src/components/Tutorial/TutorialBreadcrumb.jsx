import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import { Breadcrumb } from 'semantic-ui-react';

const TutorialBreadcrumb = props => (
  <Breadcrumb>
    <Breadcrumb.Section link as={Link} to="/">
      Home
    </Breadcrumb.Section>
    <Breadcrumb.Divider icon="right chevron" />
    <Breadcrumb.Section link as={Link} to={props.collection_path}>
      {props.collection_title}
    </Breadcrumb.Section>
    <Breadcrumb.Divider icon="right chevron" />
  </Breadcrumb>
);

TutorialBreadcrumb.propTypes = {
  collection_title: PropTypes.string,
  collection_path: PropTypes.string,
};

export default TutorialBreadcrumb;
