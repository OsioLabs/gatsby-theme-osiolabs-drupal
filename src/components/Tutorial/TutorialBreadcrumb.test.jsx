import React from 'react';
import { render } from '@testing-library/react';

import TutorialBreadcrumb from './TutorialBreadcrumb';

describe('Component: <TutorialBreadcrumb />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(
      <TutorialBreadcrumb
        collection_title="Tacos Rock"
        collection_path="/collection/tacos"
      />
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
