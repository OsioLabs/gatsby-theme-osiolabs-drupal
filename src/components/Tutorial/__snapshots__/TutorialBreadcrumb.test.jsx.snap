// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TutorialBreadcrumb /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui breadcrumb"
  >
    <a
      class="section"
      href="/"
    >
      Home
    </a>
    <i
      aria-hidden="true"
      class="right chevron icon divider"
    />
    <a
      class="section"
      href="/collection/tacos"
    >
      Tacos Rock
    </a>
    <i
      aria-hidden="true"
      class="right chevron icon divider"
    />
  </div>
</DocumentFragment>
`;
